Sources/ConcurrencyHelpers/Event.swift

```diff
@@ -0,0 +1,93 @@
+import Atomics
+
+/**
+ * Synchronization primitive modelled after Event from WinAPI.
+ */
+public final class Event {
+    // Protects continuations array and write to signaled flag.
+    private var lock = Lock()
+
+    // Can be accessed for read without taking lock
+    private var signaled: UnsafeAtomic<Bool>
+
+    private typealias Continuation = CheckedContinuation<Void, Never>
```

Comment:
Since it's generic implementation in helpers I would consider add Success and Error as part of Event generic declaration. Then signal can return some object or throw an error

Reply:
I think then it will be `Future` :)

Reply:
it could be, but then we will do the break :slightly_smiling_face: [](edited-with-Axolo)

Reply:
but maybe it’s okay for now [](edited-with-Axolo)